### ISSUE TITLE
FATAL EXCEPTION: Thread-2

### ISSUE THREAD

**Issue by gnusagit** [vavadiyahiren/xmpp_plugin#2]
when i call *await xmppConnection.login();* function. this in my code:

```
Future<void> _startXmpp() async {
  final XmppConnection xmppConnection = XmppConnection(<String, String>{
    'user_jid':'<username>',
    'password': '<password>',
    'host': '<host>',
    'port': '<port>',
  });
  await xmppConnection.start((dynamic event) async { //onReceivedMessage
    log('testOnMessage');
    log(event.toString());
  }, (dynamic event) async { //onError
    log('testError');
    log(event.toString());
  });
  await xmppConnection.login();
}
```

and this is my console:

```
D/loginTest(31557): onMethodCall call: login
D/flutter_xmpp(31557): onCreate()
D/flutter_xmpp(31557): onStartCommand()
D/flutter_xmpp(31557):  Service Start() function called.
D/flutter_xmpp(31557): initConnection()
D/flutter_xmpp(31557): Connection Constructor called.
E/AndroidRuntime(31557): FATAL EXCEPTION: Thread-2
E/AndroidRuntime(31557): Process: id.gnusa.bluebutterfly.blue_butterfly, PID: 31557
E/AndroidRuntime(31557): java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
E/AndroidRuntime(31557): 	at org.xrstudio.xmpp.flutter_xmpp.FlutterXmppConnection.<init>(FlutterXmppConnection.java:83)
E/AndroidRuntime(31557): 	at org.xrstudio.xmpp.flutter_xmpp.FlutterXmppConnectionService.initConnection(FlutterXmppConnectionService.java:94)
E/AndroidRuntime(31557): 	at org.xrstudio.xmpp.flutter_xmpp.FlutterXmppConnectionService.access$100(FlutterXmppConnectionService.java:16)
E/AndroidRuntime(31557): 	at org.xrstudio.xmpp.flutter_xmpp.FlutterXmppConnectionService$1.run(FlutterXmppConnectionService.java:120)
E/AndroidRuntime(31557): 	at java.lang.Thread.run(Thread.java:923)
D/OOMEventManagerFK(31557): checkEventAndDumpForJE: 0
I/Process (31557): Sending signal. PID: 31557 SIG: 9
Lost connection to device.
```

**Comment by vavadiyahiren** [comment]
@gnusagit Can you please pass username alongwith the resource name to check if that works with you.

I will fix the problem with the new version soon.

**Comment by vavadiyahiren** [comment]
Fixed with the latest version v1.1.14
